### ISSUE TITLE
Pump suspend is turned off by iAPS without time expiry or user change

### ISSUE THREAD

**Issue by JELCRAWFORD** [Artificial-Pancreas/iAPS#485]
Suspending the pump initially results in suspend behavior and the omnipod will beep in confirmation and pump will show as suspended.  However this suspend will not necessarily last for the requested duration (10, 60, 90 or 120 minutes) selected in the settings.  It is not clear what provoked the pump suspend to be turned off but it did seem to happen in conjunction with a loop running.  If the pump has been suspended this behaviour is undesirable and potentially harmful, especially if it has been suspended due to low BG or other issues where the user expects there to be zero insulin delivery.  Either the time should expire or the user should have to manually have to toggle the pump back on.  In our case, neither of these things happened.  Logs attached.
[crawford_log.txt](https://github.com/Artificial-Pancreas/iAPS/files/13983495/crawford_log.txt)
[crawford_log_prev.txt](https://github.com/Artificial-Pancreas/iAPS/files/13983497/crawford_log_prev.txt)

software version v.2.3.0

**Comment by Jon-b-m** [comment]
How did you suspend. Via iAPS?

**Comment by Jon-b-m** [comment]
I’ll need a timestamp of suspension.

**Comment by Jon-b-m** [comment]
I need more info here. When did you suspend pod (timestamp), how did you suspend the pod(on pump or in iAPS), for how long did you suspend the pod and when was the pod unexpectedly unsuspended?

**Comment by JELCRAWFORD** [comment]
Hi Jon - this problem occurred at the same time as the unexpected behaviour with insulin delivery that I also entered as an issue.  I had thought I’d put the logs in for both issues, there should be a timestamp in the logs.  The pump suspension was used via the iAPS app to stop any insulin delivery when DS was very low; however, it did not stay suspended. I can only suspend the pump through app as it is an omnipod.

I can not recall at this point the length of time of the suspend, but I believe it was probably an hour.  I am not clear on what caused it to be unsuspended, but it was not the time running out on the suspend, in some cases it was unsuspending itself in less than 10 minutes after it was set.  It recurred several times over the time period of the low, from about 1amMST to 4 or 5am MST.  I am unsure now how many times that was attempted.  I should be able to copy the logs in from the other issue, do you need me to do that?

Thx,
Jennifer

**Comment by Jon-b-m** [comment]
There are 16MBs of logs, which is why a more exact timestamp and descriptive issue report would help a lot. 

From just looking at the logs I can see that a pump suspension was created at 02:11 and the Pump was later resumed at 02:41. That's 30 min. 

During this time no insulin was enacted. 

A new pump suspension was created at 02:50. Pump was resumed at 03:20. That's 30 min. 

During this time no insulin was enacted.

**Comment by Jon-b-m** [comment]
30 min is the first duration in the time picker list. Looks like you picked the first for every suspension?

**Comment by Jon-b-m** [comment]
Since you’re not commenting I guess this issue is cleared?

**Comment by JELCRAWFORD** [comment]
Sorry, I did not see a request for a comment, but I find Discord difficult to navigate, so likely my bad.  I believe the problem of the unsuspend problem is because we had toggled on the "unsuspend if no temp” from our Medtronic days.  This should probably be a part of the instructions for that specific feature that Omnipod users to turn it off, as there is no useful function, and in fact, if you believe you have suspended the omnipod, it will not stay that way if there is no temp  (in Medtronic days we’d have unclipped it so this would not have been a problem).  

Questions welcome if you don’t understand the sequence of events.
